oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/NimbusJwtDecoderJwkSupport.java

```diff
@@ -104,17 +112,25 @@ public NimbusJwtDecoderJwkSupport(String jwkSetUrl, String jwsAlgorithm) {
 			new JWSVerificationKeySelector<>(this.jwsAlgorithm, jwkSource);
 		this.jwtProcessor = new DefaultJWTProcessor<>();
 		this.jwtProcessor.setJWSKeySelector(jwsKeySelector);
+
+		// Spring Security validates the claim set independent from Nimbus
+		this.jwtProcessor.setJWTClaimsSetVerifier((claims, context) -> {});
 	}
 
 	@Override
 	public Jwt decode(String token) throws JwtException {
 		JWT jwt = this.parse(token);
 		if (jwt instanceof SignedJWT) {
-			return this.createJwt(token, jwt);
+			Jwt createdJwt = this.createJwt(token, jwt);
+			return this.validateJwt(createdJwt);
 		}
 		throw new JwtException("Unsupported algorithm of " + jwt.getHeader().getAlgorithm());
 	}
 
+	public void setJwtValidator(OAuth2TokenValidator<Jwt> jwtValidator) {
```

Comment:
Can we change this to `setValidator()` given that `Jwt` is implied with the generic

Reply:
Let's see if we can get on the same page about this with @rwinch. Rob, I remember you mentioning that you thought we ought to move the dsl method `decoder` to `jwtDecoder`. If we go that route, it seems consistent to do the same with validators.

Truth be told, I'm not sure I understand the reason why `jwtDecoder()` is preferrable to `decoder()`--do you think it applies here, too, Rob?